cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -372,8 +373,36 @@ Future<Void> rollingUpdateForNewCaKey() {
         }
     }
 
-    // ZooKeeper is rolled only for new Cluster CA key
-    private Future<Void> maybeRollZookeeper(RestartReasons podRestartReasons) {
+    /**
+     * Finds the number of current ZooKeeper replicas. We need to find the current number because if new trust should be
+     * rolled out while scaling event is happening, we have to roll out the trust only to the existing pods.
```

Comment:
can you be clear about what's the "trust" we are talking about here? I could think about trusting a new cluster CA but trying to understand its relation with the subject of this PR (the description doesn't mention any "trust").

Reply:
I improved the Javadoc comment here. Please have a look if it is better now.